include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -417,12 +418,13 @@ def EventOp : MooreOp<"wait_event", [
 
 def ConstantOp : MooreOp<"constant", [Pure]> {
   let summary = "A constant integer value";
-  let arguments = (ins APIntAttr:$value);
+  let arguments = (ins FVIntegerAttr:$value);
   let results = (outs IntType:$result);
   let hasCustomAssemblyFormat = 1;
   let hasVerifier = 1;
   let hasFolder = 1;
   let builders = [
+    OpBuilder<(ins "IntType":$type, "const FVInt &":$value)>,
     OpBuilder<(ins "IntType":$type, "const APInt &":$value)>,
     OpBuilder<(ins "IntType":$type, "int64_t":$value)>,
```

Comment:
From my perspective, I think `FVInt` can shoulder the creation of `moore.constant`. So should we still need `APInt`?

Reply:
I totally agree! The builder with `APInt` is just here for convenience now, it internally calls the builder with `FVInt` and converts the `APInt` to an `FVInt` without any X or Z bits.